pkg/cloud/services/ec2/bastion.go

```diff
@@ -67,6 +69,12 @@ func (s *Service) ReconcileBastion() error {
 	// Describe bastion instance, if any.
 	instance, err := s.describeBastionInstance()
 	if awserrors.IsNotFound(err) {
+		if conditions.GetReason(s.scope.AWSCluster, infrav1.BastionHostReadyCondition) == "" {
```

Comment:
Just in case there's an issue with creating an instance on AWS to avoid the flickering between `BastionCreationStartedReason` and `BastionHostFailedReason`